### DISCUSSION TITLE
Молитвы Готика 2 Классик

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2_classic#76] [General]
оказалось что ловкость за молитвы может выпасть только у Пала :-) 

```D
else if(SpecialBless <= 10)
	{
		if(hero.guild == GIL_PAL)
		{
			if(zufall >= 3)
			{
				B_BlessAttribute(hero,ATR_MANA_MAX,4);
			}
			else if(zufall >= 2)
			{
				B_BlessAttribute(hero,ATR_HITPOINTS_MAX,8);
			}
			else if(zufall >= 1)
			{
				B_BlessAttribute(hero,ATR_STRENGTH,2);
			}
			else if(zufall >= 0)
			{
				B_BlessAttribute(hero,ATR_DEXTERITY,2);
			};
		}
		else if(hero.guild == GIL_KDF)
		{
			B_BlessAttribute(hero,ATR_MANA_MAX,4);
		}
		else if(zufall >= 2)
		{
			B_BlessAttribute(hero,ATR_HITPOINTS_MAX,4);
		}
		else
		{
			B_BlessAttribute(hero,ATR_STRENGTH,1);
		};
		SpecialBless = SpecialBless + 1;
	}
```
в веб архиве тоже пишут что ловкости нет
![image](https://user-images.githubusercontent.com/49514764/109429297-23938a00-7a0c-11eb-9515-d924c586b533.png)

***

но при этом пишут почему то про главы :-)

else if(SpecialBless <= 10)  // получается что бонусов не 10, а  11 ? :-)

_Originally posted by @Kor-Angar in https://github.com/dosinabox/g2_classic/issues/9#issuecomment-787498802_

**Comment by Kor-Angar** [comment]
поправь меня, если я что то  не правильно понял:

перманентных бонусов 11

бонус(любой) за молитвы  можно получить 1 раз в день.

***

**при молитве 50 золотых** перманентные бонусы
паладин: перманентно +2 маны  или +4 жизни
маг: перманентно +2 маны
прочие гильдии: перманентно +2 жизни

**при молитве 100 золотых** перманентные бонусы
Паладин: перманентно +2 силы или +2 ловкости или +4 маны или +8 жизни
Маг: перманентно +4 маны
Прочие гильдии: перманентно +1 сила или +4 жизни

***

**При молитвы без пожертвования:**
Паладин восстановить  5 жизней  или 5 маны
Маг восстановить  5 маны
эффект такой же как еды / растений, т.е. не перманентный.

**при молитве 10 золотых, если все 11 перманентных бонусов уже были получены**
Паладин  полностью восстановить жизни и ману
Маг пополнить ману на 5.
эффект такой же как еды / растений, т.е. не перманентный.

**при молитве 50 золотых, если все 11 перманентных бонусов уже были получены**
все гильдии: восстановить жизнь на 5

**при молитве 100 золотых, если все 11 перманентных  бонусов  уже были получены**
Паладин  полностью восстановить жизни и ману
Маг полностью восстановить жизни и ману
прочие гильдии: восстановить 5 жизни

**Reply by Kor-Angar** [reply]
неправильно

**Comment by Kor-Angar** [comment]
@dosinabox  проверь пожалуйста по коду бонусы за блогославление, так ли я расписал

Обычны благословения герой так же получает раз в день.
**Паладин**

- При молитве без жертвы золота Паладин восстановит или жизни или мана. Шанс на восстановление жизней выше.
- При молитве за 10 золотых Паладину восстановятся и жизни и мана.
- При молитве за 50 Паладину, после получения перманентных бонусов, будут восстановлены только жизни :-(
- При молитве за 100 золотых Паладину. после получения перманентных бонусов, будут восстановлены и жизни и мана.

Таким образом Паладину жертвовать 50 или 100 золотых после получения перманентных бонусов нецелесообразно

Маг Огня

- При молитве без жертвы золота Маг Огня имеет шанс восстановить ману.
- При молитве за 10 золотых Магу Огня восстановится мана.
- При молитве за 50 золотых Магу Огня после получения перманентных бонусов будут восстановлены жизни.
- При молитве за 100 золотых Магу Огня после получения перманентных бонусов, будут восстановлены жизни и мана.
По мнению автору Магу Огня оптимально молится за 10 золотых.

Остальные

-  При молитве без жертвы золота бонуса не будет.
-  При молитве за 10 золотых бонуса не будет
-  При молитве за 50 золотых после получения перманентных бонусов будут восстановлены жизни.
-  При молитве за 100 золотых после получения перманентных бонусов, будут восстановлены жизни.

Оптимальным вариантом будет молитвы за 50 золотых, после получения перманентных бонусов . https://steamcommunity.com/sharedfiles/filedetails/?id=2845556136

**Reply by Kor-Angar** [reply]
@dosinabox  проверь пожалуйста

**Reply by Kor-Angar** [reply]
в интернете почти не осталось инфы по оригинальной игре. http://gothic.ag.ru отвалился, а стрые форумы где это есть частично закрыты от индексации.

**Reply by dosinabox** [reply]
Пока не могу заняться. Можешь пока изучить бэкапы gothic.ag.ru в архиве Wayback Machine: https://web.archive.org/web/20220708051415/https://gothic.ag.ru/

Крайний от 8 июля.

**Reply by Kor-Angar** [reply]
а чего мне их изучать. я же там больше половины написал :-) и там ошибки есть :-)

**Reply by Kor-Angar** [reply]
> Пока не могу заняться. 

я тебе в конце недели напомню еще раз

**Comment by Kor-Angar** [comment]
какая то непонятная проверка была на день молитвы. перенёс изменения счётчика, добавил  1 и стало как то понятнее и проще:-)
```D
func int C_HeroPrayedToday()
{
    var int xyz;    xyz = false;
    var int today;  today = Wld_GetDay() + 1 ;
    
    if (PRAYDAY == today)
    {
        PrintScreen(Print_BlessNone,-1,40,FONT_Screen,3);
        xyz = true;
    };
    
    PRAYDAY = today;
    return xyz;
}; 
```

**Reply by dosinabox** [reply]
А в нулевой день работает?

**Reply by Kor-Angar** [reply]
да

**Reply by Kor-Angar** [reply]
в нулевой день today = 1

**Reply by Kor-Angar** [reply]
я прям кусок кода туда перенёс сначала. так как он везле повторялся 
![image](https://github.com/dosinabox/g2_classic/assets/49514764/d05aa5a3-f2b9-4343-9471-2abae55a652e)

он стал какой то тут же сложный и я решил что надо переписать. полчаса въезжал и пол часа проверял :-)

**Reply by Kor-Angar** [reply]
пожалуй можно и укрупнить

```D
func int C_PrayShrine_Pray_Pay(var int PayGold)
{   
    test(Stats_Blessings_GoldGiven)             { Stats_Blessings_GoldGiven += PayGold; };
	Npc_RemoveInvItems(hero,ItMi_Gold,PayGold);
 	Info_ClearChoices(PC_PrayShrine_Pray);
    
    var int xyz;    xyz = false;
    var int today;  today = Wld_GetDay() + 1 ;
    
    if (PRAYDAY == today)
    {
        PrintScreen(Print_BlessNone,-1,40,FONT_Screen,3);
        xyz = true; // будет выход
    };
    
	PRAYDAY = today;
    return xyz;     
};
```


```D
func void PC_PrayShrine_Pray_BigPay()
{

 	if (C_PrayShrine_Pray_Pay(100))
	{
        return; // выход
    };
    
    // даём бонусы
    if (!B_BlessSpecial(100))
    {
        B_BlessNoSpecial();
    };
};
```

**Reply by Kor-Angar** [reply]
в итоге такая структура получилась

```D

func void PC_PrayShrine_Pray_BigPay()
{
    if (C_PrayShrine_Pray_Pay(100))
    {
        return; // выход
    };
    
    // даём бонусы
    if (!B_BlessSpecial(100))
    {
        B_BlessNoSpecial(100);
    };
};

func void pc_prayshrine_pray_mediumpay()
{

    if (C_PrayShrine_Pray_Pay(50))
    {
        return; // выход
    };
    
    // даём бонусы
    if (!B_BlessSpecial(50))
    {
        B_BlessNoSpecial(50);
    };  
};


func void pc_prayshrine_pray_smallpay()
{
    if (C_PrayShrine_Pray_Pay(10))
    {
        return; // выход
    };
    
    // даём бонусы
    B_BlessNoSpecial(10);
};

func void pc_prayshrine_pray_nopay()
{
    if (C_PrayShrine_Pray_Pay(0))
    {
        return; // выход
    };
    
    // даём бонусы
    B_BlessNoSpecial(0);
};
```
